app_internal_test.go

```diff
@@ -115,3 +117,37 @@ func TestAnnotationError(t *testing.T) {
 	assert.ErrorIs(t, err, wantErr)
 	assert.Contains(t, err.Error(), wantErr.Error())
 }
+
+// TestDisableSignalHandler tests disabling the default signal handler.
+// It's an internal test to allow us replace the app.receivers.notify function.
+func TestDisableSignalHandler(t *testing.T) {
+	for _, disableSignalHandler := range []bool{false, true} {
+		testName := "Not disabling signal handler"
+		if disableSignalHandler {
+			testName = "Disabling signal handler"
+		}
+		t.Run(testName, func(t *testing.T) {
+			app := New()
+			calledNotify := false
+
+			app.receivers.notify = func(c chan<- os.Signal, sig ...os.Signal) {
```

Comment:
You could have passed a func to func DisableSignalHandler() Option {

So it would be something like that

```go
func DisableSignalHandler(callback func()) Option {
```

So here you would use

```go
var notifyCalled bool
DisableSignalHandler(func () {notifiyCalled = true}
```

Because right now, you have not testing the method you added.

Reply:
Maybe there's a misunderstanding? `DisableSignalHandler()` disables fx's default use of `signal.Notify` (via `app.receivers.notify`). This tests that `DisableSignalHandler` effectively prevents fx from registering with `signal.Notify`.

A callback is not required because I'm disabling the notify registration. If I wanted to provide my own notify registration method as an option I would do as you say. However, I don't have a use case for a general purpose option. I think the specific option is a nicer api.

Reply:
OK then 👍